File: dto_maker/print_well_report/BusinessAssociate.go
package dto

type Business_associate struct {
	Business_associate_id string  `json:"business_associate_id" default:"source"`
	Active_ind            *string `json:"active_ind" default:""`
	Ba_abbreviation       *string `json:"ba_abbreviation" default:""`
	Ba_category           *string `json:"ba_category" default:""`
	Ba_code               *string `json:"ba_code" default:""`
	Ba_long_name          *string `json:"ba_long_name" default:""`
	Ba_short_name         *string `json:"ba_short_name" default:""`
	Ba_type               *string `json:"ba_type" default:""`
	Credit_check_date     *string `json:"credit_check_date" default:""`
	Credit_check_ind      *string `json:"credit_check_ind" default:""`
	Credit_check_source   *string `json:"credit_check_source" default:""`
	Credit_rating         *string `json:"credit_rating" default:""`
	Credit_rating_source  *string `json:"credit_rating_source" default:""`
	Current_status        *string `json:"current_status" default:""`
	Effective_date        *string `json:"effective_date" default:""`
	Expiry_date           *string `json:"expiry_date" default:""`
	First_name            *string `json:"first_name" default:""`
	Last_name             *string `json:"last_name" default:""`
	Middle_initial        *string `json:"middle_initial" default:""`
	Ppdm_guid             *string `json:"ppdm_guid" default:""`
	Remark                *string `json:"remark" default:""`
	Service_period        *string `json:"service_period" default:""`
	Source                *string `json:"source" default:""`
	Row_changed_by        *string `json:"row_changed_by" default:""`
	Row_changed_date      *string `json:"row_changed_date" default:""`
	Row_created_by        *string `json:"row_created_by" default:""`
	Row_created_date      *string `json:"row_created_date" default:""`
	Row_effective_date    *string `json:"row_effective_date" default:""`
	Row_expiry_date       *string `json:"row_expiry_date" default:""`
	Row_quality           *string `json:"row_quality" default:""`
}

File: create_tables/business_associate.py
import oracledb
import sys
import lib_platform as platform

if platform.system == "darwin":
    sys.path.insert(0, "database")
elif platform.system == "windows":
    sys.path.insert(0, "../database")

from conn import connection

# pw = getpass.getpass("Enter password: ")

# connection = oracledb.connect(
#     user="admin",
#     password="ppdm-3.9",
#     dsn="/kei-oracle-test.chkxr21l2l5s.ap-southeast-3.rds.amazonaws.com:1521/ORCL")

# print("Successfully connected to Oracle Database")

cursor = connection.cursor()


cursor.execute("""
    begin
        execute immediate 'drop table business_associate';
        exception when others then if sqlcode <> -942 then raise; end if;
    end;""")

cursor.execute(
    """create table business_associate(
    BUSINESS_ASSOCIATE_ID VARCHAR(40) PRIMARY KEY NOT NULL,
    ACTIVE_IND VARCHAR(1),
    BA_ABBREVIATION VARCHAR(12),
    BA_CATEGORY VARCHAR(40),
    BA_CODE VARCHAR(40),
    BA_LONG_NAME VARCHAR(255),
    BA_SHORT_NAME VARCHAR(30),
    BA_TYPE VARCHAR(40),
    CREDIT_CHECK_DATE TIMESTAMP(0),
    CREDIT_CHECK_IND VARCHAR(1),
    CREDIT_CHECK_SOURCE VARCHAR(40),
    CREDIT_RATING VARCHAR(40),
    CREDIT_RATING_SOURCE VARCHAR(40),
    CURRENT_STATUS VARCHAR(40),
    EFFECTIVE_DATE TIMESTAMP(0),
    EXPIRY_DATE TIMESTAMP(0),
    FIRST_NAME VARCHAR(255),
    LAST_NAME VARCHAR(255),
    MIDDLE_INITIAL VARCHAR(30),
    PPDM_GUID VARCHAR(38),
    REMARK VARCHAR(2000),
    SERVICE_PERIOD VARCHAR(240),
    SOURCE VARCHAR(40),
    ROW_CHANGED_BY VARCHAR(30),
    ROW_CHANGED_DATE TIMESTAMP(0),
    ROW_CREATED_BY VARCHAR(30),
    ROW_CREATED_DATE TIMESTAMP(0),
    ROW_EFFECTIVE_DATE TIMESTAMP(0),
    ROW_EXPIRY_DATE TIMESTAMP(0),
    ROW_QUALITY VARCHAR(40))""")